### DISCUSSION TITLE
Waiver acquisition dollar amount

### DISCUSSION THREAD

**Discussion by jon-cheng** [cwendt94/espn-api#170] [ESPN API] [answered]
Thanks for building this, it's extremely useful and fun to play around with FF league data! I was wondering if there's a way to extract the FAAB dollars spent on waiver wire acquisitions. I see that there is ADD/DROP data as well as bid amount for auction drafts, but was not able to get the waiver acquisition dollar amount. Thanks for your help!

**Comment by cwendt94** [comment, marked as answer]
Great question! I have looked at this before but at that time I was not apart of any FAAB leagues and couldn't see the data where they set the bid amount. This past year I joined a FAAB league and now have the data! It should be an easy update to the [Activity Object](https://github.com/cwendt94/espn-api/wiki/Activity-Class)

**Reply by jon-cheng** [reply]
Thank you! I look forward to that update.

**Reply by cwendt94** [reply]
`bid_amount` is now part of [Activity Object](https://github.com/cwendt94/espn-api/wiki/Activity-Class) in package version 0.10.0

**Reply by chieftainSupreme** [reply]
did bid_amount disappear?  not seeing it come through but it's still listed on the documentation

**Reply by cwendt94** [reply]
That logic has not changed, so it should still be there unless ESPN changed where the data lives

**Reply by eggman920** [reply]
I'm having issues finding the FAAB info as well. I can get player acquisition type but can get the amount spent if they were added. anyone have any tips, (using python)

**Reply by cwendt94** [reply]
Are you looking at the `recent_activity` functionality or directly in the player class? Recent activity should have bid amount. For the player class it currently doesn't have any field for bid amount/FAAB. Not sure if ESPN stores that data directly with the player. Do you have an example on ESPN's website showing the FAAB amount?

**Reply by eggman920** [reply]
<img width="782" alt="Screen Shot 2024-08-21 at 5 11 53 PM" src="https://github.com/user-attachments/assets/bb72f680-8818-449a-bb43-cf121ceb68ed">

I found some addition info as well here is the actual url, request url and a preview of the response

https://fantasy.espn.com/football/league/offerreport?leagueId=<leagueid>

https://lm-api-reads.fantasy.espn.com/apis/v3/games/ffl/seasons/2023/segments/0/leagues/<leagueid>?scoringPeriodId=2&view=mDraftDetail&view=mStatus&view=mSettings&view=mTeam&view=mTransactions2&view=modular&view=mNav

sorry for the json mess

`transactions
: 
[{bidAmount: 2, executionType: "PROCESS", id: "d45f13af-98ce-48e5-a15c-91b2b24552ae",…},…]
0
: 
{bidAmount: 2, executionType: "PROCESS", id: "d45f13af-98ce-48e5-a15c-91b2b24552ae",…}
bidAmount
: 
2
executionType
: 
"PROCESS"
id
: 
"d45f13af-98ce-48e5-a15c-91b2b24552ae"
isActingAsTeamOwner
: 
false
isLeagueManager
: 
false
isPending
: 
false
items
: 
[{fromLineupSlotId: -1, fromTeamId: 0, isKeeper: false, overallPickNumber: 0, playerId: -16019,…}]
0
: 
{fromLineupSlotId: -1, fromTeamId: 0, isKeeper: false, overallPickNumber: 0, playerId: -16019,…}
fromLineupSlotId
: 
-1
fromTeamId
: 
0
isKeeper
: 
false
overallPickNumber
: 
0
playerId
: 
-16019
toLineupSlotId
: 
20
toTeamId
: 
10
type
: 
"ADD"
memberId
: 
"NightlyLeagueUpdateTaskProcessor"
processDate
: 
1694590231927
proposedDate
: 
1694590231846
rating
: 
0
relatedTransactionId
: 
"27b6658d-1e80-46f9-b1c3-c8b94a303ab4"
scoringPeriodId
: 
2
skipTransactionCounters
: 
false
status
: 
"EXECUTED"
subOrder
: 
22
teamId
: 
10
type
: 
"WAIVER"
1
: 
{bidAmount: 13, executionType: "PROCESS", id: "d89b4d48-6094-44eb-8f54-67ac4770c3b6",…}
2
: 
{bidAmount: 0, executionType: "PROCESS", id: "9b7c371e-4efc-4b6d-b51e-0faa4a91994d",…}
3
: 
{bidAmount: 0, executionType: "PROCESS", id: "21f08e92-bb74-4717-beba-bdc8d26ae925",…}
4
: 
{bidAmount: 0, executionType: "PROCESS", id: "9302419e-a6d6-4e83-81b5-246962c1cccd",…}
5
: 
{bidAmount: 12, executionType: "PROCESS", id: "5ebe09c6-96cb-4aab-ad9f-4d3d10a34eb4",…}
6
: 
{bidAmount: 23, executionType: "PROCESS", id: "7de68830-3b7d-4148-a6a8-70f15a0f5571",…}
7
: 
{bidAmount: 15, executionType: "PROCESS", id: "e7335b91-54bc-48b4-981a-b096920e9966",…}
8
: 
{bidAmount: 41, executionType: "PROCESS", id: "1c9cba69-84c2-421c-8f0c-a35ebdd1ff37",…}
9
: 
{bidAmount: 0, executionType: "CANCEL", id: "0ddf7b6b-9115-4e65-a4c0-3ee9fd67b998",…}
10
: 
{bidAmount: 7, executionType: "PROCESS", id: "ba56ab06-5b23-487e-893c-760d8e2e64fc",…}
11
: 
{bidAmount: 53, executionType: "PROCESS", id: "40c04ae9-554b-4dfc-933e-6adb824d9c21",…}
12
: 
{bidAmount: 33, executionType: "PROCESS", id: "9c642c3a-3113-4a1c-9bd4-cf1eebb25860",…}
13
: 
{bidAmount: 13, executionType: "PROCESS", id: "e954d7a4-be6f-47d2-89e6-53671b0343da",…}
14
: 
{bidAmount: 29, executionType: "PROCESS", id: "e5cf4cbe-3f0c-4b8b-a55e-abedb58435dc",…}
15
: 
{bidAmount: 1, executionType: "PROCESS", id: "e7f0588d-5f73-4066-8547-bbad7075ad1a",…}
16
: 
{bidAmount: 12, executionType: "PROCESS", id: "e4f71539-545c-4c87-bea9-28cecdb482ac",…}
17
: 
{bidAmount: 2, executionType: "PROCESS", id: "1d093f19-a9aa-403f-bb5f-e7e473820cdd",…}
18
: 
{bidAmount: 3, executionType: "PROCESS", id: "ea677b48-e45b-4144-9102-2050a3b6bb73",…}
19
: 
{bidAmount: 13, executionType: "PROCESS", id: "69bd9a9b-f26e-4bf4-b6b9-3e72c92407b0",…}
20
: 
{bidAmount: 2, executionType: "PROCESS", id: "9f3527ee-693d-486e-9c79-bba7003c13cc",…}
21
: 
{bidAmount: 11, executionType: "PROCESS", id: "4af57c71-0cfd-4d70-aefa-dbe8ff484d8b",…}
22
: 
{bidAmount: 12, executionType: "PROCESS", id: "0b091507-2a20-464d-bfb0-982adc8ee392",…}
23
: 
{bidAmount: 29, executionType: "PROCESS", id: "f220e8eb-cacd-484d-9ad1-54242b4aee55",…}
24
: 
{bidAmount: 1, executionType: "PROCESS", id: "d03f9595-d89d-4754-95b6-41943034dcf2",…}
25
: 
{bidAmount: 8, executionType: "PROCESS", id: "eb048dd7-a129-4854-ae6f-fb3b02efcaa4",…}
26
: 
{bidAmount: 22, executionType: "PROCESS", id: "7e484a64-e20a-4cc6-b956-85eb01703d6a",…}
27
: 
{bidAmount: 8, executionType: "PROCESS", id: "a1908b22-96ed-42be-b3ed-cee7a076e45d",…}
28
: 
{bidAmount: 10, executionType: "PROCESS", id: "c9dc9e37-0428-4080-a2b9-b23137eb20a5",…}`

**Reply by cwendt94** [reply]
Oh okay, this looks like you can get the whole leagues transaction history. Playing around with `view=mTransactions2` I see they put all the different types of transactions including roster moves, wavier, bids and trades. Of course the format is nothing like recent activity lol. I can see adding a `get_transactions` functionality in the league class

**Reply by simonbilskyrollins** [reply]
I went ahead and opened a PR that implements a method to fetch transactions for basketball leagues: https://github.com/cwendt94/espn-api/pull/598

The code should be easily portable over to football leagues as well based on the info I see in this thread. Hope this helps!

**Reply by cwendt94** [reply]
Thanks @simonbilskyrollins!
